harness/determined/pytorch/tensorboard.py

```diff
@@ -0,0 +1,55 @@
+# mypy: ignore-errors
+# This is to silence mypy. All SummaryWriter methods are untyped, but are inherited here.
+import pathlib
+from typing import Union
+
+from determined import tensorboard
+
+# As of torch v1.9.0, torch.utils.tensorboard has a bug that is exposed by setuptools 59.6.0.  The
+# bug is that it attempts to import distutils then access distutils.version without actually
+# importing distutils.version.  We can workaround this by prepopulating the distutils.version
+# submodule in the distutils module.
+import distutils.version  # isort:skip  # noqa: F401
+from torch.utils.tensorboard import SummaryWriter  # isort:skip
+
+
+class TorchWriter(SummaryWriter, tensorboard.MetricWriter):
```

Comment:
Don't hate me, but after our last round of discussion regarding how we decided we can't do a lot to help CoreAPI+PyTorch users, is there any point in even having this object?

Thinking about this:
- PyTorchTrial users: context.get_tensorboard_writer() returns a plain SummaryWriter, which we call `.close()` on when we want to.  Done.
- Pytorch + Core API users: they create a SummaryWriter and they have to set the log_dir with `core_context.training.get_tensorboard_path()`, and they also have to call `.close()` when they need to.  Not great but also having our own subclass of SummaryWriter clearly doesn't make anything easier.
- Core API + TensorBoardMode.AUTO: this can go through the old TorchWriter path as before.  Well... you'd probably want to do something very slightly different so you could give an accurate deprecation warning to users trying to use the old TorchWriter directly:
```python
# in det.tensorboard.metric_writers.pytorch

class _TorchWriter(...):
    ...

class TorchWriter(_TorchWriter):
    def __init__(self, *args, **kwargs):
        warnings.warn("this is deprecated, ...", FutureWarning, stacklevel=2)
        super().__init__(*args, **kwargs)
```